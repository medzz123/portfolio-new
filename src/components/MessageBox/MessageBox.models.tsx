export interface MessageBoxProps {
  type?: 'error' | 'info';
}
