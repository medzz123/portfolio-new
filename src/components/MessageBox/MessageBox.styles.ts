import { styled } from '@theme/config';

export const MessageBoxContainer = styled('div', {
  backgroundColor: '$error',
  borderRadius: '$sizes$12',
  padding: '$12',
  width: 'fit-content',

  variants: {
    type: {
      info: {
        backgroundColor: '$primary',
      },
      error: {
        backgroundColor: '$error',
      },
    },
  },
});
