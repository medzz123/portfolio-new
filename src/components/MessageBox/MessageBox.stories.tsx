import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';

import MessageBox from './MessageBox';

export default {
  title: 'Components/MessageBox',
  component: MessageBox,
} as Meta;

const Template: Story = (storyArguments) => <MessageBox {...storyArguments} />;

export const Default = Template.bind({});
