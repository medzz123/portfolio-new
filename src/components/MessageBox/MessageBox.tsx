import Text from '@components/Text';
import React, { FunctionComponent } from 'react';

import { MessageBoxProps } from './MessageBox.models';
import { MessageBoxContainer } from './MessageBox.styles';

const MessageBox: FunctionComponent<MessageBoxProps> = (props) => {
  const { type = 'info', children } = props;

  return (
    <MessageBoxContainer type={type} data-testid="messageBox">
      <Text mb="none" level="info">
        {children}
      </Text>
    </MessageBoxContainer>
  );
};

export default MessageBox;
