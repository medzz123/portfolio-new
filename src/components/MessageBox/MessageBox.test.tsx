import { render } from '@test/library';
import React from 'react';

import MessageBox from './MessageBox';

test('MessageBox Test', () => {
  const { getByTestId, container } = render(<MessageBox>Hello</MessageBox>);

  const messageBoxNode = getByTestId('messageBox');

  expect(messageBoxNode.textContent).toBe('Hello');

  expect(container).toMatchSnapshot();
});
