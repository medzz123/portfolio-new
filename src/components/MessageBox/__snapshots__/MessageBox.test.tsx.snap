// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox Test 1`] = `
<div>
  <div
    class="sxeg1y8 sxeg1y880y8k--type-info"
    data-testid="messageBox"
  >
    <p
      class="sx8ivet sx8ivet59c0g--level-info sx8iveth4u4f--weight-normal sx8ivetfxkwo--color-normal sx8ivet0o7ux--align-left sx8ivetzvy3i--spacing-none sx8ivet8316w--font-mulish"
      data-testid="text"
    >
      Hello
    </p>
  </div>
</div>
`;
